core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1829,7 +1901,58 @@ private[spark] class BlockManager(
     data.dispose()
   }
 
+  /**
+   * Class to handle block manager decommissioning retries
+   * It creates a Thread to retry offloading all RDD cache blocks
+   */
+  private class BlockManagerDecommissionManager(conf: SparkConf) {
+    @volatile private var stopped = false
+    private val sleepInterval = conf.get(
+      config.STORAGE_DECOMMISSION_REPLICATION_REATTEMPT_INTERVAL)
+
+    private val blockReplicationThread = new Thread {
+      override def run(): Unit = {
+        var failures = 0
+        while (blockManagerDecommissioning
+          && !stopped
+          && !Thread.interrupted()
```

Comment:
Do we really need this? 

1. In `stop()`, `stopped` is set before `interrupt()`. 

2. if the thread is doing decommission or sleeping when `stop()` is called, we'll capture `InterruptedException` later and set `stopped` to true.

Reply:
Unless the interrupt exception is caught inside of the block transfer

Reply:
> Unless the interrupt exception is caught inside of the block transfer

Reply:
> Unless the interrupt exception is caught inside of the block transfer

If there's an `InterruptedException` captured by block transfer, then `Thread.interrupted()` inside `blockReplicationThread` would return false. And what do you expect for this case?

If you want the decommission thread stop, then, `Thread.interrupted()` won't work;

Or if you want the decommission to keep working, then, `Thread.interrupted()` is useless because the status has already been cleared(unless block transfer set it to interrupted again).

Reply:
If an interrupt exception is caught the thread would still be marked as interrupted

Reply:
I believe in normal cases like `wait`, `sleep`, the status will be cleared according to JDK doc:

```
* @throws  InterruptedException
*          if any thread has interrupted the current thread. The
*          <i>interrupted status</i> of the current thread is
*          cleared when this exception is thrown.
```

And even if the status is not be cleared in other cases, the following `Thread.sleep(sleepInterval)` will throw `InterruptedException` firstly and set `stopped` to true and `Thread.interrupted()` still does not take effect.

Reply:
So that block of text indicates that the interrupted status is cleared when the exception is thrown. Which means it would be possible for the thread to be in an interrupted status without having the exception thrown. Worst case scenario: this check is not needed but also causes no significant harm